src/java.base/share/classes/java/lang/invoke/MethodHandleImpl.java

```diff
@@ -2089,21 +1875,90 @@ static Object tryFinally(MethodHandle target, MethodHandle cleanup, Object... av
         return r;
     }
 
+    // see varargsArray method for chaching/package-private version of this
+    private static MethodHandle makeCollector(Class<?> arrayType, int parameterCount) {
+        MethodType type = MethodType.methodType(arrayType, Collections.nCopies(parameterCount, arrayType.componentType()));
+
+        MethodHandle newArray = MethodHandles.arrayConstructor(arrayType);
+        MethodHandle storeFunc = ArrayAccessor.getAccessor(arrayType, ArrayAccess.SET);
```

Comment:
I'd move `storeFunc` into `makeCollectorForm()` and use erased accessor there (`ArrayAccessor.OBJECT_ARRAY_SETTER`). 

(It is interesting that `OBJECT_ARRAY_SETTER` is wrapped into `Intrinsic.ARRAY_STORE` intrinsic.)

Reply:
Done. I think it's also more obvious this way that the erased setter is being used.